docs-source/content/userguide/managing-domains/domain-lifecycle/liveness-readiness-probe-customization.md

```diff
@@ -0,0 +1,80 @@
+---
+title: "Liveness and readiness probes customization"
+date: 2019-02-23T20:58:51-05:00
+draft: false
+weight: 5
+description: "This document describes how to customize the liveness and readiness probes for WebLogic Server instance Pods."
+---
+
+This document describes how to customize the liveness and readiness probes for WebLogic Server instance Pods.
+
+#### Contents
+
+* [Liveness probe customization](#liveness-probe-customization)
+* [Readiness probe customization](#readiness-probe-customization)
+
+#### Liveness probe customization
+
+The liveness probe is configured to check that the server is alive by querying the Node Manager process.  The liveness probe is by default configured to check liveness every 15 seconds, and to timeout after 5 seconds.  If a pod fails the liveness probe, Kubernetes will restart that container. 
+
+The liveness probe interval and timeout can be customized using the `livenessProbe` attribute under the `serverPod` element of the domain resource. 
+
+Following is an example configuration to change liveness probe interval and timeout value.
+```
+  serverPod:
+    livenessProbe:
+      periodSeconds: 30
+      timeoutSeconds: 10
+```
+
+It is also possible to customize the liveness probe by specifying a custom script which will be invoked by the existing liveness probe script (`livenessProbe.sh`) after the existing script performs its own checks. This custom script can be specified either by using the `livenessProbeCustomScript` attribute in the domain resource or by setting `LIVENESS_PROBE_CUSTOM_SCRIPT` environment variable using `env` attribute under the `serverPod` element (see configuration examples below). If the custom script fails with non-zero exit status, the liveness probe will fail and Kubernetes will restart the container.
+
+**Note**: The liveness probe custom script option is for advanced usage only and its value is not set by default. If the specified script is not found, then custom script is ignored and existing liveness script will perform its normal checks.
+
+**Note**: Oracle recommends against having any long running calls (e.g. any network calls or executing wlst.sh) in the liveness probe custom script.
+
+Use the following configuration to specify liveness probe custom script using `livenessProbeCustomScript` domain resource field.
+```
+spec:
+  livenessProbeCustomScript: /u01/customLivenessProbe.sh
+```
+
+Use the following configuration to specify liveness probe custom script using `LIVENESS_PROBE_CUSTOM_SCRIPT` environment variable.
+```
+    serverPod:
+      env:
+      - name: LIVENESS_PROBE_CUSTOM_SCRIPT
+        value: /u01/customLivenessProbe.sh
+```
+
+Following environment variables are available for use in the liveness probe custom script which will be invoked by `livenessProbe.sh`.
```

Comment:
The following environment variables that are populated by the Operator are available for use in the ...